### DISCUSSION TITLE
How to solve compatibility of Proxy in v7

### DISCUSSION THREAD

**Discussion by 1995gaowei** [pixijs/pixijs#9644] [Q&A]
I found that Loader class used Proxy feature (v7.2.4 in use), but i cannot transpile or polyfill it by babel. How to solve this because i dont want to use v4 any more. And i wonder  is there any new feature cannot be transpiled like Proxy in v7.2.4?

**Comment by SuperSodaSea** [comment]
While a full Proxy polyfill is not possible, you can use <https://github.com/GoogleChrome/proxy-polyfill> as a polyfill since pixi.js only use the `set` trap in Proxy. I haven't tried it myself, but it is very likely to work.

**Reply by 1995gaowei** [reply]
I found that lib too, but im wondering if it supports proxy array perfectly? The Loader._parsers is an array. Anyway, thanks for your reply, maybe i will go back to v4 eventually.

**Reply by SuperSodaSea** [reply]
Yeah it seems that this polyfill cannot handle dynamic properties.

**Comment by sbyps** [comment]
@SuperSodaSea so can the proxy-polyfill works well with pixi7.4.2 in es5 version?

**Reply by sbyps** [reply]
i try the proxy-polyfill in chrome39, can not work

**Reply by SuperSodaSea** [reply]
Yeah, proxy-polyfill cannot handle dynamic properties, so I think the answer is no...

**Reply by sbyps** [reply]
please take a look at es6-proxy-polyfill, can this package support pixi7 proxy?
I also report a feature request for no proxy use in pixi.js, https://github.com/pixijs/pixijs/issues/10829, please consider for that
@SuperSodaSea

**Reply by SuperSodaSea** [reply]
es6-proxy-polyfill says:

> For non-array object, the properties you want to proxy must be known at creation time

So I think it don't support pixi.js either.
